### PR TITLE
handle capture syntax

### PR DESCRIPTION
**Pull request by AdiZav**
##### Short description:
Allow passing capture strings to support nmpolicy.
Capture will be passed to interface configuration, dns-resolver, routes, and capture field

### CODE REVIEW

File: ocp_resources/node_network_configuration_policy.py [hunk 1]
```diff
@@ -189,16 +194,26 @@ def add_interface(
...
         if set_ipv4:
-            iface["ipv4"] = {
-                "enabled": ipv4_enable,
-                "dhcp": ipv4_dhcp,
-                "auto-dns": ipv4_auto_dns,
-            }
-            if ipv4_addresses:
-                iface["ipv4"]["address"] = ipv4_addresses
+
+            if type(set_ipv4) is str:
```
Comment (myakove) [inline ocp_resources/node_network_configuration_policy.py:198]: please use isinstance()

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
@AdiZav please update the README with example how to use capture

File: ocp_resources/node_network_configuration_policy.py (hunk 1 above)
Comment (AdiZav) [reply ocp_resources/node_network_configuration_policy.py:198]: ACK

**Review by myakove — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -133,6 +133,21 @@ command_output = test_vmi.virt_launcher_pod.execute(command="command-example")
 ```
 If no Pod was found, will raise ``ResourceNotFoundError``.
 
+### NNCP Capture Syntax
+Using capture syntax to switch ipv4 config between interfaces
+```python
+with NodeNetworkConfigurationPolicy(
+    name = "capture_nncp",
```
Comment (myakove) [inline README.md:140]: please remove spaces before and after `=`

File: ocp_resources/node_network_configuration_policy.py [hunk 3]
```diff
@@ -307,8 +324,9 @@ def wait_for_interface_deleted(self):
     def validate_create(self):
         for pod in self.worker_pods:
             for bridge in self.ifaces:
-                node_network_state = NodeNetworkState(name=pod.node.name)
-                node_network_state.wait_until_up(name=bridge["name"])
+                if "capture" not in bridge["name"]:
```
Comment (myakove) [inline ocp_resources/node_network_configuration_policy.py:327]: Please explain why

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_policy.py (hunk 3 above)
Comment (AdiZav) [reply ocp_resources/node_network_configuration_policy.py:327]: When configuring an interface whose name is used with capture, like in: 
``- name: "{{ capture.br1.interfaces.0.bridge.port.0.name }}"
      type: ethernet
      state: up
      ipv4: "{{ capture.br1.interfaces.0.ipv4 }}"
      ipv6: "{{ capture.br1.interfaces.0.ipv6 }}"``
     
py-wrapper checks if interface {{ capture.br1.interfaces.0.bridge.port.0.name }} is up, there's no way to pass/know actual name

File: README.md (hunk 2 above)
Comment (AdiZav) [reply README.md:140]: ACK

File: ocp_resources/node_network_configuration_policy.py (hunk 3 above)
Comment (myakove) [reply ocp_resources/node_network_configuration_policy.py:327]: Please add comment about it in the code.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_policy.py (hunk 3 above)
Comment (AdiZav) [reply ocp_resources/node_network_configuration_policy.py:327]: Sure

File: ocp_resources/node_network_configuration_policy.py [hunk 4]
```diff
@@ -307,8 +324,11 @@ def wait_for_interface_deleted(self):
     def validate_create(self):
         for pod in self.worker_pods:
             for bridge in self.ifaces:
-                node_network_state = NodeNetworkState(name=pod.node.name)
-                node_network_state.wait_until_up(name=bridge["name"])
+                if (
+                    "capture" not in bridge["name"]
+                ):  # Wrapper can't tell a capture iface's actual name
```
Comment (myakove) [inline ocp_resources/node_network_configuration_policy.py:329]: don't we set the name?
capture={'first-nic': 'interfaces.name=="ens8"',
              'second-nic': 'interfaces.name=="ens9"'},

Comment (AdiZav) [reply ocp_resources/node_network_configuration_policy.py:329]: That was just a simple example. Usually, like when taking over a bridge's old port, you won't know the port name.
I.e, capture={..., 'interface': 'capture.br1.interfaces.0.bridge.port.0.name'}

File: ocp_resources/node_network_configuration_policy.py [hunk 5]
```diff
@@ -454,7 +456,7 @@ def _get_nnce_error_msg(nnce_name, nnce_condition):
 
         libnmstate_err = re.findall(r"libnmstate.error.*", nnce_msg)
         if libnmstate_err:
-            err_msg += f"{nnce_prefix }: {libnmstate_err[0]}"
+            err_msg += f"{nnce_prefix}: {libnmstate_err[0]}"
```
Comment (AdiZav) [inline ocp_resources/node_network_configuration_policy.py:459]: Edited by black

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**